boards/arm/bbc_microbit/bbc_microbit.dts

```diff
@@ -22,3 +22,7 @@
 	status = "ok";
 	current-speed = <115200>;
 };
+
+&i2c0 {
+	status = "ok";
```

Comment:
I would propose to add `clock-frequency = <I2C_BITRATE_FAST>;` for all boards in the PR because most users will use 400 kHz bitrate. If you don't, just keep in mind that every user should create a board overlay to override it in application level.